frontend/src/concepts/pipelines/content/configurePipelinesServer/ConfigurePipelinesServerModal.tsx

```diff
@@ -92,8 +104,62 @@ export const ConfigurePipelinesServerModal: React.FC<ConfigurePipelinesServerMod
     configureDSPipelineResourceSpec(configureConfig, project.metadata.name)
       .then((spec) => {
         createPipelinesCR(namespace, spec)
-          .then(() => {
+          .then((obj: DSPipelineKind) => {
             onBeforeClose();
+
+            const pollingNamespace = obj.metadata.namespace;
+            notification.info(`Waiting on pipeline server resources for ${pollingNamespace}...`);
+            registerNotification({
+              callbackDelay: FAST_POLL_INTERVAL || 3000,
+              callback: async (signal: AbortSignal) => {
+                try {
+                  // check if polling for too long
+                  if (
+                    Date.now() - new Date(obj.metadata.creationTimestamp || Date.now()).getTime() >
+                    SERVER_TIMEOUT
+                  ) {
+                    throw Error(`${pollingNamespace} pipeline server creation timed out`);
+                  }
```

Comment:
We have https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/concepts/pipelines/context/usePipelineNamespaceCR.ts#L14 to detect the time out, try to reuse this function. You can refer to the usage in `PipelinesContext` file.